File: backend/src/main/java/ca/metricalsky/winston/entity/VideoEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;

@Entity
@Table(name = "videos")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VideoEntity {

    @Id
    @Column(name = "id")
    private String id;

    @Basic(optional = false)
    @Column(name = "channel_id")
    private String channelId;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @Column(name = "published_at")
    private OffsetDateTime publishedAt;

    @UpdateTimestamp
    @Column(name = "last_fetched_at")
    private OffsetDateTime lastFetchedAt;

    @OneToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "id", referencedColumnName = "video_id")
    private VideoCommentsEntity comments;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/video/VideoEntity.kt
package ca.metricalsky.winston.database.entity.video

import jakarta.persistence.*
import org.hibernate.annotations.UpdateTimestamp
import java.time.OffsetDateTime

@Entity
@Table(name = "videos")
class VideoEntity(

    @Id
    @Column(name = "id")
    var id: String? = null,

    @Basic(optional = false)
    @Column(name = "channel_id")
    var channelId: String? = null,

    @Column(name = "title")
    var title: String? = null,

    @Column(name = "description")
    var description: String? = null,

    @Column(name = "thumbnail_url")
    var thumbnailUrl: String? = null,

    @Column(name = "published_at")
    var publishedAt: OffsetDateTime? = null,
) {
    @UpdateTimestamp
    @Column(name = "last_fetched_at")
    var lastFetchedAt: OffsetDateTime? = null

    @OneToOne(cascade = [CascadeType.PERSIST, CascadeType.MERGE])
    @JoinColumn(name = "id", referencedColumnName = "video_id")
    var details: VideoDetailsEntity? = null

    @OneToOne(cascade = [CascadeType.PERSIST])
    @JoinColumn(name = "id", referencedColumnName = "video_id")
    var comments: VideoCommentsEntity? = null
}
